### PR TITLE
Add consensus threshold param to oracle contract

### PR DESCRIPTION
**Pull request by denalimarsh**
Closes: https://github.com/cosmos/peggy/issues/150

This PR adds a consensus threshold parameter to the Oracle contract so that the threshold is configurable like it is in the oracle module.

Instead of hardcoding the threshold at `(signed power*3) >= (total power*2)`, the threshold is now set at `signed power >= x% of total power` where `x` is the consensus threshold parameter. For example, if the consensus threshold is 65, all prophecies will require 65% of the total available validator power in order to be processed.
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: testnet-contracts/migrations/2_next.js [hunk 1]
```diff
@@ -81,7 +90,7 @@ module.exports = function(deployer, network, accounts) {
     // 1. Deploy BridgeToken contract
     //    Gas used:        1,884,394 Gwei
     //    Total cost:    0.03768788 Ether
-    await deployer.deploy(BridgeToken, "TEST", {
+    await deployer.deploy(BridgeToken, "stake", {
```
Comment (okwme) [inline testnet-contracts/migrations/2_next.js]: why is this called stake? isn't this just a dummy ERC-20 to be used as an Eth based asset that is moved to the cosmos chain?

File: testnet-contracts/contracts/Oracle.sol [hunk 2]
```diff
@@ -241,15 +248,15 @@ contract Oracle {
                 }
         }
 
-        // Calculate if weighted signed power has reached threshold of weighted total power
-        uint256 weightedSignedPower = signedPower.mul(3);
-        uint256 weightedTotalPower = totalPower.mul(2);
-        bool hasReachedThreshold = weightedSignedPower >= weightedTotalPower;
+        // Prophecy must reach total signed power % threshold in order to pass consensus
```
Comment (okwme) [inline testnet-contracts/contracts/Oracle.sol]: Maybe note that the `consensusThreshold` is a decimal multiplied by 100 which is why the `signedPower` is also being multiplied by 100

File: testnet-contracts/migrations/2_next.js (hunk 1 above)
Comment (denalimarsh) [reply testnet-contracts/migrations/2_next.js]: Ah, good catch. While attempting to end-to-end test around the issues with whitelisting native cosmos assets described in https://github.com/cosmos/peggy/issues/123 I updated the testing token's denom to stake. Fixed token denom in https://github.com/cosmos/peggy/pull/160/commits/b5b5d7d47d7531d2b7420dcb8da450a37bb917aa.

The whitelisting bug will be addressed within a few days as part of updating namespacing.

File: testnet-contracts/contracts/Oracle.sol (hunk 2 above)
Comment (denalimarsh) [reply testnet-contracts/contracts/Oracle.sol]: https://github.com/cosmos/peggy/pull/160/commits/75f75ed2b188a9b9d9527ecd70ab4330cf4970b4

**Comment by okwme** [comment]
Made https://github.com/denalimarsh/peggy/pull/3 to hopefully fix issue running tests in CI

**Comment by denalimarsh** [comment]
@okwme looks like this is cleared for merge

**Pull request merged**